### PR TITLE
Add vertical sharing card

### PR DESCRIPTION
**Pull request by MiSikora**
## Description

Adds a vertical card variant for sharing. Card image isn't `200 dp` like in the designs but instead 70% of the card size and card size is determined by the aspect ratio of 2:3. This is done to accommodate for different screen sizes.

Podcast card design: HR2vcQrWcS0scsolKZBitg-fi-2203_2520
Episode card design: HR2vcQrWcS0scsolKZBitg-fi-2203_2368

## Testing Instructions

Code and preview review should be enough but you can check this card with clip sharing in the portrait orientation.

## Screenshots or Screencast 

Screenshot 2024-07-24 at 15 03 45

## Checklist
- [x] ~If this is a user-facing change, I have added an entry in CHANGELOG.md~
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [x] ~I have considered whether it makes sense to add tests for my changes~
- [x] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [x] Any jetpack compose components I added or changed are covered by compose previews
- [x] ~I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.~
 
#### I have tested any UI changes...

- [x] with different themes
- [x] with a landscape orientation
- [ ] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">This PR is tagged with <code>do not merge</code> label(s).</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: modules/features/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/ui/VerticalCard.kt [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+                modifier = Modifier.size(18.dp),
+            )
+            Spacer(
+                modifier = Modifier.width(8.dp),
+            )
+            TextH70(
+                text = "Pocket Casts",
+                color = Color.White,
+            )
+        }
```
Comment (ashiagr) [inline modules/features/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/ui/VerticalCard.kt:143]: Can we make this pill a reusable component?

File: modules/features/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/ui/VerticalCard.kt [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+            Image(
+                painter = painterResource(id = R.drawable.ic_logo_foreground),
+                contentDescription = null,
+                modifier = Modifier.size(18.dp),
+            )
+            Spacer(
+                modifier = Modifier.width(8.dp),
+            )
+            TextH70(
+                text = "Pocket Casts",
```
Comment (ashiagr) [inline modules/features/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/ui/VerticalCard.kt:140]: It doesn't matter, but can we reuse the untranslatable string resource for the app title?

**Review by ashiagr — COMMENTED** [review]
In case they're not on your radar:

1. I cannot see the clipper at the bottom of the card on the tablet:




2. Card background gradient seems to mingle with the image background, there's not enough contrast between them:

In Figma | Actual
---|----
 |

File: modules/features/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/ui/VerticalCard.kt (hunk 2 above)
Comment (MiSikora) [reply modules/features/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/ui/VerticalCard.kt:140]: Yes, totally makes sense!

File: modules/features/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/ui/VerticalCard.kt (hunk 1 above)
Comment (MiSikora) [reply modules/features/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/ui/VerticalCard.kt:143]: Sure, I'll extract it.

**Comment by MiSikora** [comment]
> I cannot see the clipper at the bottom of the card on the tablet:

Oh wow, something broke. I'll investigate.

> Card background gradient seems to mingle with the image background, there's not enough contrast between them:

We don't have colors specification ATM. I asked for it here pdeCcb-6h3-p2

**Comment by MiSikora** [comment]
> I cannot see the clipper at the bottom of the card on the tablet:

@ashiagr The issue seems to be with the clip page. Forcing a 3:2 aspect ratio on the card can cause problems on different devices, especially when there's hard-coded padding. I plan to work on a new iteration for the sharing page designs, as I need to add support for podcasts and episode sharing anyway. Would you mind if I address it then?

File: modules/features/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/ui/VerticalCard.kt (hunk 2 above)
Comment (MiSikora) [reply modules/features/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/ui/VerticalCard.kt]: Changed in https://github.com/Automattic/pocket-casts-android/pull/2518/commits/6239ca294153bd92f108e214524799adbd1e55db

File: modules/features/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/ui/VerticalCard.kt (hunk 1 above)
Comment (MiSikora) [reply modules/features/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/ui/VerticalCard.kt]: Change in https://github.com/Automattic/pocket-casts-android/pull/2518/commits/a08263f10250915650219101d6452562c67cb033

**Review by ashiagr — APPROVED** [review]
Thanks for the updates!
Feel free to address tablet issue separately.

**Pull request merged**